<launch>
  <arg name="input_device_enabled" default="true" />
  <!-- Choose witch interface is used. 
  For now, only one interface at a time is supported -->
  <arg name="xbox" default="false" />
  <arg name="xbox_dev" default="js0" />
  <arg name="spacenav" default="false" />
  <arg name="webapp" default="true" />

  <group if="$(arg input_device_enabled)">
    <group if="$(arg xbox)">
        <param name="joy_node/deadzone" value="0.2" type="double"/>
        <param name="joy_node/coalesce_interval" value="0.01" type="double"/>
        <param name="joy_node/autorepeat_rate" value="50" type="double"/>
        <param name="joy_node/dev" value="/dev/input/$(arg xbox_dev)" type="string"/>
        <node name="joy_node" pkg="joy" type="joy_node" output="screen" respawn="false" />
        <node name="device_xbox" pkg="orthopus_space_control" type="device_xbox" output="screen" respawn="false">  
            <rosparam file="$(find orthopus_space_control)/config/xbox_settings.yaml"/>
        </node>
    </group>

    <group if="$(arg spacenav)">
        <node name="spacenav_node" pkg="spacenav_node" type="spacenav_node" output="screen" respawn="false">
          <rosparam param="static_rot_deadband">0.5</rosparam>
          <rosparam param="static_trans_deadband">0.5</rosparam>
        </node>
        <node name="device_spacenav" pkg="orthopus_space_control" type="device_spacenav" output="screen" respawn="false">  
            <rosparam file="$(find orthopus_space_control)/config/spacenav_settings.yaml"/>
        </node>
    </group>
    
    <group if="$(arg webapp)">
        <node name="device_web_app" pkg="orthopus_space_control" type="device_web_app" output="screen" respawn="false"/> 
        <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" output="screen" respawn="true"/>
        <node pkg="nodelet" type="nodelet" name="libuvc_camera" args="load libuvc_camera/driver /nodelet_manager" output="screen" respawn="true">
          <param name="frame_id" value="hand_link" />
          <!-- Parameters used to find the camera -->
          <param name="vendor" value="0x045e"/>
          <param name="product" value="0x0810"/>
          <!-- Image size and type -->
          <param name="width" value="640"/>
          <param name="height" value="360"/>
          <param name="video_mode" value="mjpeg"/>
          <param name="frame_rate" value="15"/>
        </node>
        <node pkg="web_video_server" type="web_video_server" name="web_video_server" output="screen"/>
    </group>
  </group>
  
  <node name="cartesian_controller" pkg="orthopus_space_control" type="cartesian_controller" output="screen" respawn="false" >
    <rosparam file="$(find orthopus_space_control)/config/settings.yaml"/>
  </node>
</launch>
